crypto/context.c

```diff
@@ -666,3 +667,19 @@ CRYPTO_THREAD_LOCAL *ossl_lib_ctx_get_rcukey(OSSL_LIB_CTX *libctx)
         return NULL;
     return &libctx->rcu_local_key;
 }
+
+int OSSL_LIB_CTX_get_conf_diagnostics(OSSL_LIB_CTX *libctx)
+{
+    libctx = ossl_lib_ctx_get_concrete(libctx);
+    if (libctx == NULL)
+        return 0;
+    return libctx->conf_diagnostics;
+}
+
+void OSSL_LIB_CTX_set_conf_diagnostics(OSSL_LIB_CTX *libctx, unsigned int value)
+{
+    libctx = ossl_lib_ctx_get_concrete(libctx);
+    if (libctx == NULL)
+        return;
+    libctx->conf_diagnostics = value != 0;
```

Comment:
Does this need to be locked/atomic?  Or can we guarantee that it isn't sensible to access in a threading situation?

It's a bit field so every access could require multiple instructions.  At least, this is true when writing, reading is less obvious.  Alternatively, make it not a bit field and atomic.

Reply:
I do not think this particular value needs to be locked but there is a good point about the bit field. I'll change that.